devShell.nix

```diff
@@ -7,9 +7,61 @@ let
   cachixName = cachixMetadata.name or null;
   cachixKey = cachixMetadata.key or null;
 
+  # Get all the options' name declared immediately under `config.devshell` by
+  # devshell's modules.
+  devshellOptions = lib.filterAttrs
+    (_: lib.isType "option")
+    (pkgs.devshell.eval { configuration = {}; }).options.devshell;
+
+  # A helper function moving all options defined in the root of the config
+  # (which name matches ones in `devshellOptions`) under a `devshell` attribute
+  # set in the resulting config.
+  #
+  # devshellOptions = { foo = ...; bar = ...; };
+  # pushUpDevshellOptions { foo = "foo"; baz = "baz"; }
+  # -> { devhsell.foo = "foo"; baz = "baz"; }
+  #
+  # Issues a warning if it would override an exisiting option:
+  #
+  # pushUpDevshellOptions { foo = "foo"; devshell.foo = "oof"; }
+  # -> { devhsell.foo = "foo"; }
+  # trace: warning: Option 'foo' defined twice, both under 'config' and
+  #   'config.devshell'. This likely happens when defining both in `Cargo.toml`:
+  #   ```toml
+  #   [workspace.metadata.nix.devshell]
+  #   name = "example"
+  #   [workspace.metadata.nix.devshell.devshell]
+  #   name = "example"
+  #   ```
+  pushUpDevshellOptions = config: let
+    movedOpts = lib.flip lib.filterAttrs config (name: _:
+      lib.warnIf
+        (lib.hasAttr name config.devshell)
```

Comment:
This will result in an error if `config.devshell` does not exist. Should be changed to `(config.devshell or { })`.

Reply:
I think `config.devshell` is guaranteed to exist since at least one sub-option has a default value (actually I think all do) ?

A quick check in the nix-repl also suggests that:
```nix
((import <nixpkgs> { overlays = [ (import ("${builtins.getFlake "github:numtide/devshell"}" + "/overlay.nix")) ]; }).devshell.eval { configuration = {}; }).config.devshell
```

Reply:
I ran it without the fix on one of my projects and Nix told me "attribute doesn't exist", so I guess it may not exist?

Reply:
Yeah my bad, the config here is before module evaluation, so it may not exist indeed.